README.md

```diff
@@ -1,7 +1,334 @@
+```csharp
+PushDocument verySecureDocument = new PushDocument("http://www.coveo.com/verysecure") {
+    ClickableUri = "http://www.coveo.com/verysecure",
+    ModifiedDate = DateTime.UtcNow
+};
+
+DocumentPermissionSet adminSet = new DocumentPermissionSet {
+    AllowedPermissions = new List<PermissionIdentity> {
+        new PermissionIdentity(@"acme\admin", PermissionIdentityType.User)
+    }
+};
+
+DocumentPermissionLevel adminLevel = new DocumentPermissionLevel {
+    PermissionSets = new List<DocumentPermissionSet> { adminSet }
+};
+
+DocumentPermissionSet usersSet = new DocumentPermissionSet {
+    AllowedPermissions = new List<PermissionIdentity> {
+        new PermissionIdentity(@"acme\john", PermissionIdentityType.User)
+    },
+    DeniedPermissions = new List<PermissionIdentity> {
+        new PermissionIdentity(@"acme\cannotaccess", PermissionIdentityType.User)
+    }
+};
+
+DocumentPermissionLevel userLevel = new DocumentPermissionLevel {
+    PermissionSets = new List<DocumentPermissionSet> { usersSet }
+};
+
+verySecureDocument.Permissions.Add(adminLevel);
+verySecureDocument.Permissions.Add(userLevel);
+
+client.DocumentManager.AddOrUpdateDocument(sourceId, verySecureDocument, null);
+```
+
+Then, you can push your identities as we did before.
+```csharp
+PermissionIdentity allowedMember = new PermissionIdentity(@"acme\john", PermissionIdentityType.User);
+PermissionIdentity deniedMember = new PermissionIdentity(@"acme\cannotaccess", PermissionIdentityType.User);
+PermissionIdentity admin = new PermissionIdentity(@"acme\admin", PermissionIdentityType.User);
+
+PermissionIdentityBody bodyAllowedMember = new PermissionIdentityBody(allowedMember);
+bodyAllowedMember.Mappings.Add(new PermissionIdentity("john@acme.com", PermissionIdentityType.User));
+
+PermissionIdentityBody bodyDeniedMember = new PermissionIdentityBody(deniedMember);
+bodyDeniedMember.Mappings.Add(new PermissionIdentity("cannotaccess@acme.com", PermissionIdentityType.User));
+
+PermissionIdentityBody bodyAdmin = new PermissionIdentityBody(admin);
+bodyAdmin.Mappings.Add(new PermissionIdentity("admin@acme.com", PermissionIdentityType.User));
+
+List<PermissionIdentityBody> mappingsMembersToAddOrUpdate = new List<PermissionIdentityBody> {
+    bodyAllowedMember,
+    bodyDeniedMember,
+    bodyAdmin
+};
+
+// Push user mappings and group members in one batched operation.
+client.PermissionManager.AddOrUpdateIdentities(expandedProviderId, null, mappingsMembersToAddOrUpdate);
+```
+
+## Activate logging
+The SDK uses `log4net` as its logging library. It can be useful to have some logs in case of a problem. In your log4net configuration, add a logger with `Coveo` as the name. We use namespaces as logger name. Using `Coveo` as the logger name will get you all the logs from the SDK. 
```

Comment:
"to have some logs in case of a problem" -> "to have logs in case of problems"
"with `Coveo` as the name" -> "named `Coveo`"
"We use namespaces as logger name" -> "We use namespaces as logger names" (I think)

Reply:
```suggestion
The SDK uses `log4net` as its logging library. It can be useful to have logs in case of problems. In your log4net configuration, add a logger with named `Coveo`. We use namespaces as logger names. Using `Coveo` as the logger name will get you all the logs from the SDK. 
```